src/app/backend/resource/controller/controller.go

```diff
@@ -251,18 +267,38 @@ func (self StatefulSetController) UID() types.UID {
 }
 
 // GetLogSources is an implementation of the GetLogSources method from ResourceController interface.
-func (self StatefulSetController) GetLogSources(k8sClient *client.Clientset) LogSources {
+func (self StatefulSetController) GetLogSources(k8sClient *client.Clientset) (LogSources, error) {
```

Comment:
I'm wondering why isn't this done same as for `Get` method. I mean that we don't need to pass client here. List of pods can already be retrieved in `getLogSourceFromController` method and passed here to do resource specific handling (filter by owner ref or filter pods for job).